core/platform/auth/firebase_auth_services.py

```diff
@@ -136,6 +128,39 @@ def authenticate_request(request):
     return None if not auth_id else auth_domain.AuthClaims(auth_id, email)
 
 
+def delete_auth_associations(user_id):
+    """Deletes associations referring to the given user_id.
+
+    Args:
+        user_id: str. The ID of the user being deleted.
+    """
+    assoc_model = (
+        auth_models.UserIdByFirebaseAuthIdModel.get_by_user_id(user_id))
+    if assoc_model is None:
+        return
+    try:
+        with _acquire_firebase_context():
```

Comment:
Seeing the usage of `_acquire_firebase_context` being approx the same everywhere would it make sense to use it as decorator for the functions where it is needed?

Reply:
Hmm for some functions yes, but limiting the scope allows us to avoid calling the SDK as much as possible (whereas otherwise we always have to call the SDK before even entering the function), so I'd prefer to keep it this way so we can return early sooner.